turftopic/models/keynmf.py

```diff
@@ -114,13 +123,16 @@ def extract_keywords(
         for i in range(total):
             terms = document_term_matrix[i, :].todense()
             embedding = embeddings[i].reshape(1, -1)
-            nonzero = terms > 0
-            if not np.any(nonzero):
-                keywords.append(dict())
-                continue
-            important_terms = np.squeeze(np.asarray(nonzero))
-            word_embeddings = self.vocab_embeddings[important_terms]
-            sim = cosine_similarity(embedding, word_embeddings)
+            if self.keyword_scope == 'document':
+                nonzero = terms > 0
+                if not np.any(nonzero):
+                    keywords.append(dict())
+                    continue
+                important_terms = np.squeeze(np.asarray(nonzero))
+                word_embeddings = self.vocab_embeddings[important_terms]
+                sim = cosine_similarity(embedding, word_embeddings)
+            else:
+                sim = cosine_similarity(embedding, self.vocab_embeddings)
```

Comment:
I don't know if you tested this code, but my guess is that it wouldn't work, cause `important_terms` is unbound if `self.keyword_scope == corpus` and it's getting used later in the code. (A good linter would catch something like this for you btw, we should also add integration tests in CI to catch this sorta stuff)

`important_terms` describes in this case which terms are the ones in the vocabulary that are relevant for a given document (as a boolean mask over the vocabulary). Since in this case all words are relevant we could do something like:
```python
n_vocab = document_term_matrix.shape[1]
important_terms = np.ones(n_vocab, dtype=bool)
```

Reply:
It seems to me that you did!! Thanks for the PR, keep it up!

Reply:
this should be addressed now. also realized that there is a possibility that users pass vectorizers with vocabularies including terms that may never occur in the dictionary (e.g., if `min_df=0`, or custom vocabularies) -- [7247618](https://github.com/x-tabdeveloping/turftopic/pull/22/commits/7247618edd9e416eda9e02fde5d703354553f98d) should exclude those when `keyword_scope='corpus'`